### DISCUSSION TITLE
After switching org, studio should select first available personal layout

### DISCUSSION THREAD

**Discussion by amacneil** [foxglove/community#565] [Q&A]
**Description**
After switching org, Studio forgets that we have an active layout.

We should select the first personal layout (if there is one) if the app starts with no layout and personal layouts are available.

<img width="1535" alt="image" src="https://user-images.githubusercontent.com/637671/184020845-292d0032-5099-4c78-a96d-e44ef9f34bd1.png">

- Version: main
- OS: web
- Data source (e.g. bag file, mcap file, rosbridge, ROS 1/2 native): data platform

**Comment by snosenzo** [comment]
Looks like it's updating the url with a proper last layout for the new org that you've switched to, but it's not reflecting that in the page itself. So an easy way to reproduce this issue is to just have an invalid layoutId in the url and load the page.

**Comment by snosenzo** [comment]
oh actually, it looks like the `PlayerManager` component is specifically updating the URL with the layout whose name matches the `sourceName` of the sample data. so it doesn't update the url from whats input in any other case. looks like just setting the selectedLayoutId is only updating the URL though and not propagating through to the actual layout

**Comment by snosenzo** [comment]
So @foxymiles and I have discussed that this might not be worth fixing right now with a less wholistic solution given that past changes to initializing state from the url have broken things in production and that this seems like a fairly small corner case that doesn't have a very big impact currently. I do have a working solution (from what I can tell, just need to fix tests) on the related PR, but it will probably only be temporary as there will be a greater effort to re-architect handling of organizations across the suite. Let me know if this is important and I should continue working on it given some of the risks
